### ISSUE TITLE
How to Resume Trained Best.pt Model

### ISSUE THREAD

**Issue by HakanOZDABAK** [ultralytics/ultralytics#1983]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello everyone, I am using YOLOV8 for my graduate project so that, sometimes I have to add new classes or redesign classes images. So I need to continue train my already trained best.pt model. How can I do this? I searched and I used "resume" code but didn't work. Can someone explain this please?

**Comment by glenn-jocher** [comment]
@HakanOZDABAK this is the YOLOv8 default use case, just start training from a pretrained model:

```python
from ultralytics import YOLO

model = YOLO('path/to/your/model.pt')
results = model.train(data='data.yaml')
```

See https://docs.ultralytics.com/modes/train for details.

**Comment by HakanOZDABAK** [comment]
> @HakanOZDABAK this is the YOLOv8 default use case, just start training from a pretrained model:
> 
> ```python
> from ultralytics import YOLO
> 
> model = YOLO('path/to/your/model.pt')
> results = model.train(data='data.yaml')
> ```
> 
> See https://docs.ultralytics.com/modes/train for details.

In this data.yaml, I need all classes for continue or just new classes enough? Example: I trained tomato potato then I wanna add egg. So this data.yaml include tomato potato egg or just egg for resume?

**Comment by glenn-jocher** [comment]
@HakanOZDABAK yes if you have a previous training that finished then you can do `model = YOLO('path/to/best.pt')` to start your new training. 

The new dataset can be anything, i.e. VOC, COCO, Argoverse etc, it does not have to share a domain with best.pt. If you are asking about training on the prior data with additional data then you must train on all classes you want the new model to be able to detect. It will forget any old information unless it is again trained on it.

**Comment by HakanOZDABAK** [comment]
> @HakanOZDABAK yes if you have a previous training that finished then you can do `model = YOLO('path/to/best.pt')` to start your new training. 
> 
> The new dataset can be anything, i.e. VOC, COCO, Argoverse etc, it does not have to share a domain with best.pt. If you are asking about training on the prior data with additional data then you must train on all classes you want the new model to be able to detect. It will forget any old information unless it is again trained on it. 
> 
> 

Thanks for your all helps. Have a nice work.

**Comment by glenn-jocher** [comment]
You're welcome, @HakanOZDABAK, I'm glad I could assist you! If you have any more questions or concerns, please don't hesitate to let me know. For more YOLOv8 help and documentation, please visit https://docs.ultralytics.com. Good luck with your project!
